src/mono/mono/mini/mini.c

```diff
@@ -92,6 +92,10 @@ static gint64 discarded_jit_time;
 #define mono_jit_unlock() mono_os_mutex_unlock (&jit_mutex)
 static mono_mutex_t jit_mutex;
 
+// used for Swift interop
+GENERATE_TRY_GET_CLASS_WITH_CACHE (swift_error, "System.Runtime.InteropServices.Swift", "SwiftError")
+GENERATE_TRY_GET_CLASS_WITH_CACHE (swift_self, "System.Runtime.InteropServices.Swift", "SwiftSelf")
+
```

Comment:
you declare these in `metadata/marshal.h`, and define them in `mini/mini.c` - I think the definitions should be in `metadata/`, too.

Reply:
I suggest defining them in `marshal.c` then, and they should be accessible from `mini-<arch>.c`.